assets/js/analyzer.js

```diff
@@ -157,81 +157,82 @@ function analyze() {
     if(currentAnalyzerRequest) {
         currentAnalyzerRequest.abort();
     }
-    currentAnalyzerRequest = $.jsonp({
-        url: config.APY_URL + '/analyze',
-        pageCache: true,
-        beforeSend: ajaxSend,
-        complete: function () {
-            ajaxComplete();
-            currentAnalyzerRequest = undefined;
-        },
-        data: {
-            'lang': analyzerMode,
-            'q': $('#morphAnalyzerInput').val()
-        },
-        success: function (data) {
-            var regex = /([^<]*)((<[^>]+>)*)/g;
-            $('#morphAnalyzerOutput').empty();
-            for(var i = 0; i < data.length; i++) {
-                var leftTD = $('<td class="text-right"></td>');
-                var strong = $('<strong></strong>').text(data[i][1].trim());
-                var arrow = $('<span></span>').html('&nbsp;&nbsp;&#8620;');
-                leftTD.append(strong).append(arrow);
-
-                var rightTD = $('<td class="text-left"></td>');
-                var splitUnit = data[i][0].split('/');
-
-                if(splitUnit[1][0] === '*') {
-                    rightTD.addClass('text-danger');
-                }
 
-                var tr = $('<tr></tr>').append(leftTD).append(rightTD);
-                $('#morphAnalyzerOutput').append(tr);
-
-                var joinedMorphemes = {};
-                for(var j = 1; j < splitUnit.length; j++) {
-                    var unit = splitUnit[j];
-                    if(unit.match(regex).length > 2) {
-                        var matches = unit.match(regex);
-                        for(var k = 1; k < matches.length - 1; k++) {
-                            if(joinedMorphemes[matches[k]]) {
-                                joinedMorphemes[matches[k]].push(unit);
-                            }
-                            else {
-                                joinedMorphemes[matches[k]] = [unit];
-                            }
-                        }
+    var request = {
+        'lang': analyzerMode,
+        'q': $('#morphAnalyzerInput').val()
+    };
+    var analyzeObject = {
+        'request': request,
+        'success': handleAnalyzeSuccessResponse,
+        'error': handleAnalyzeErrorResponse,
+        'requestName': 'currentAnalyzerRequest'
+    };
+    apyCall(analyzeObject, '/analyze');
```

Comment:
Why not inline everything? The variables are extraneous if you never use them again.

```
apyCall({
    'request': {
        'lang': analyzerMode,
        'q': $('#morphAnalyzerInput').val()
    },
    'success': handleAnalyzeSuccessResponse,
    'error': handleAnalyzeErrorResponse,
    'requestName': 'currentAnalyzerRequest'
}, '/analyze');
```